xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/InlineEditableContentBlock.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.block;
+
+import java.util.List;
+
+import org.xwiki.rendering.listener.MetaData;
+import org.xwiki.stability.Unstable;
+
+/**
+ * This block represents a part of a macro content which can be edited inline.
+ *
+ * @version $Id$
+ * @since 10.9RC1
+ */
+@Unstable
+public class InlineEditableContentBlock extends MetaDataBlock
```

Comment:
I feel that the naming and documentation should reflect something a little less WYSIWYG specific: the fact that the source of this block is not modified during the macro execution.

Reply:
The fact that some block is not modified during execution doesn't mean that we want to make this content editable inline. For example you can have some static text that you don't want to be editable inline.